### PR TITLE
CLID-647: Add testing docs and skills

### PR DESCRIPTION
**Pull request by adolfo-ab**
# Description

  In order to improve our testing and make it more efficient, this PR adds a few testing-related skills and docs:

  - .agents/skills/
    - automate-integration-test: Converts a manual test case description into a Ginkgo v2 integration test, following project conventions (runner methods, assertion helpers, test images).
    - check-test-coverage: Analyzes oc-mirror CLI feature coverage across integration and e2e tests, producing a gap report with prioritized recommendations.
    - migrate-e2e-to-integration: Migrates an e2e test case to the integration suite, translating framework patterns (registry, invocation, assertions) and presenting a detailed plan before implementation.
  - docs/testing/
    - README.md: Documents the oc-mirror testing strategy — testing levels (unit, integration, e2e), their purpose, and guidelines.
  - .agents/commands/
 - Moved from .claude/commands/ to .agents/commands/ to consolidate all agent configuration under .agents/.

Github / Jira issue: 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code Improvements (Refactoring, Performance, CI upgrades, etc)
- [x] Internal repo assets (diagrams / docs on github repo)
- [ ] This change requires a documentation update on openshift docs

# How Has This Been Tested?
N/A

## Expected Outcome
N/A

## Summary by CodeRabbit

* **Documentation**
  * Renamed and updated agent command docs to use role-specific wording and updated example prompts.
  * Added a skill for automating integration-test creation with step-by-step guidance.
  * Added a skill to assess CLI test coverage and generate gap reports.
  * Added a skill for migrating E2E tests into integration tests with migration rules.
  * Added a testing strategy README covering test types, environments, and principles.

### CODE REVIEW

File: .agents/skills/automate-integration-test/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+- **Correctness**: assertions actually verify the scenario, no false positives or dead steps
+- **Efficiency**: no redundant operations, unnecessary variables, or repeated lookups
+- **Cleanliness**: no superfluous comments, unused variables, or boilerplate that adds nothing - every line should earn its place
+
+### 9. Run the tests
+
+Ask permission, then run. The `registry` binary must be in PATH - ask the user for its location if unknown:
+```bash
+PATH=$PATH:<path-to-registry-bin> go test -v ./tests/integration/ --ginkgo.focus "<test label>"
+```
```
Comment (aguidirh) [inline .agents/skills/automate-integration-test/SKILL.md]: Is a binary for the registry mandatory here? 

How about adding an alternative method running a container image ? Something like the command below: 

```
podman run -dt   -p 5000:5000   -v /home/aguidi/go/src/github.com/aguidirh/oc-mirror/alex-tests/local-registry/registry-config.yml:/etc/distribution/config.yml:z   --name local-registry   docker.io/distribution/distribution:3
```

File: docs/testing/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+Tests that focus on verifying the core logic, behavior and external dependencies of the oc-mirror CLI by using a local registry.
+
+These are black-box tests that do not have direct access to the `oc-mirror` internal code.
+
+All the images used in these tests are built by us, in order to control their content and size.
+
+In these tests we assert the tangible outputs of `oc-mirror`: archive contents, mirrored images present on the target registry, etc.
+
+> **Example:** Execute the oc-mirror CLI command to successfully copy one image from a source registry to a local registry, and then verify the image exists in the local registry.
```
Comment (aguidirh) [inline docs/testing/README.md]:
```suggestion
> **Example:** Execute the oc-mirror CLI command to successfully copy one image from a source registry to a target registry, and then verify the image exists in the target registry.
```

Instead of target it could be destination as well.

**Review by aguidirh — COMMENTED** [review]
Thanks for adding these skills! This is something that can help us in our test strategy.

I added few comments, let me know if you have questions about them.

File: docs/testing/README.md (hunk 2 above)
Comment (adolfo-ab) [reply docs/testing/README.md]: Done

File: .agents/skills/automate-integration-test/SKILL.md (hunk 1 above)
Comment (adolfo-ab) [reply .agents/skills/automate-integration-test/SKILL.md]: It was needed before, but now we're using an in-process registry, I just forgot to remove this bit. Fixed.

File: docs/testing/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+### ⬅️ Early testing/Shift-left
+
+Catch defects as early as possible in the development cycle. The earlier a bug is found, the cheaper it is to fix.
+
+### 📐 Testing Pyramid
+
+Follow the testing pyramid: many unit tests at the base, fewer integration tests in the middle, and the fewest E2E tests at the top. Each level should test what it is best suited for — don't push scenarios up the pyramid unnecessarily.
+
+```
```
[the remark this answers is not in the document]
Comment (adolfo-ab) [reply docs/testing/README.md]: Done

File: .agents/skills/migrate-e2e-to-integration/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+The user provides one of:
+- An e2e case number (e.g., 72973)
+- A test description or keyword
+
+## Step-by-step process
+
+### 1. Read project conventions
+
+Read `AGENTS.md` at the project root, and the the documents at `docs/testing` for architecture, conventions, and pitfalls.
```
[the remark this answers is not in the document]
Comment (adolfo-ab) [reply .agents/skills/migrate-e2e-to-integration/SKILL.md]: Done

File: .agents/skills/check-test-coverage/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+- It's only tested in the happy path but has error handling worth verifying
+- It's a flag that modifies behavior but is only tested implicitly (e.g., the flag is set but its effect isn't asserted)
+
+### 5. Produce the report
+
+Present results in this format:
+
+#### Summary table
+
+```
```
[the remark this answers is not in the document]
Comment (adolfo-ab) [reply .agents/skills/check-test-coverage/SKILL.md]: Done

File: .agents/skills/automate-integration-test/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+If the test case is ambiguous, ask before writing code:
+- Which mirror mode? (`mirrorToMirror`, `mirrorToDisk` + `diskToMirror`, or both)
+- Does it include a delete workflow (phase 1 + phase 2)?
+- Is a new ISC or DISC YAML needed, or does an existing one suffice?
+- Should the test go in an existing file or a new one?
+
+### 3. Check for duplicate or superfluous coverage
+
+Read existing test files to ensure the scenario isn't already covered. If it is, tell the user and suggest what would add value instead.
+Also consider if one of the existing tests could be extended to cover the scenario, without blurrying the testing boundaries.
```
[the remark this answers is not in the document]
Comment (adolfo-ab) [reply .agents/skills/automate-integration-test/SKILL.md]: Done

**Review by aguidirh — APPROVED** [review]
Thanks for adding this context to oc-mirror repo.

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Comment by aguidirh** [comment]
/assign @nidangavali 

For learning purposes only.

**Comment by aguidirh** [comment]
/verified bypass

Since there is no production code being changed, there is no need to verify.
